habitat/core/batched_env.py

```diff
@@ -15,6 +15,85 @@
 
 import torch  # isort:skip # noqa: F401  must import torch before importing bps_pytorch
 
+from habitat.utils.geometry_utils import  quaternion_rotate_vector 
+from habitat.tasks.utils import cartesian_to_polar
+
+from habitat_sim.utils.common import  quat_from_magnum
+
+
+class StateSensorConfig:
+    def __init__(self, config_key, shape, obs_key):
+        self.config_key = config_key
+        self.shape = shape
+        self.obs_key = obs_key
```

Comment:
```suggestion
@attr.s(auto_attribs=True, slots=True)
class StateSensorConfig:
    config_key : str
    self.shape : int
    self.obs_key : str
```
Then you just add the @attr.s decorator to the subclasses and you won't have to supply your own `__init__` method.